app/code/community/Bolt/Boltpay/Block/Checkout/Boltpay.php

```diff
@@ -189,185 +195,268 @@ public function getCartDataJs($checkoutType = 'multi-page')
-                    $success
-                    
-                    var input = document.createElement('input');
-                    input.setAttribute('type', 'hidden');
-                    input.setAttribute('name', 'bolt_reference');
-                    input.setAttribute('value', transaction.reference);
-                    document.getElementById('edit_form').appendChild(input);
-                    
-                    // order and order.submit should exist for admin            
-                    if ((typeof order !== 'undefined' ) && (typeof order.submit === 'function')) {
-                        order_completed = true;
-                        callback();
-                    }
-                }";
-                $onClose = "if (order_completed && (typeof order !== 'undefined' ) && (typeof order.submit === 'function')) {
-                    $close
-                    var bolt_hidden = document.getElementById('boltpay_payment_button');
-                    bolt_hidden.classList.remove('required-entry');
-                    order.submit();
-                 }";
-            } else {
-                $onSuccessCallback = "function(transaction, callback) {
-                    new Ajax.Request(
-                        '$saveOrderUrl',
-                        {
-                            method:'post',
-                            onSuccess:
-                                function() {
-                                    $success
-                                    order_completed = true;
-                                    callback();
-                                },
-                            parameters: 'reference='+transaction.reference
-                        }
-                    );
-                }";
-                $onClose = "if (typeof bolt_checkout_close === 'function') {
-                    // used internally to set overlay in firecheckout
-                    bolt_checkout_close();
-                 }
-                 if (order_completed) {
-                    location.href = '$successUrl';
-                 }";
-            }
+                bolt_hidden.classList.add('required-entry');
+                return is_valid;
+            }";
+        }
+
+        return $checkCallback;
+    }
 
-            return ("
-                var json_cart = $jsonCart;
-                var quote_id = '{$immutableQuoteId}';
-                var order_completed = false;
-                
-                BoltCheckout.configure(
-                    json_cart,
-                    $jsonHints,
+    /**
+     * @param string $successCustom
+     * @param $checkoutType
+     * @return string
+     */
+    public function buildOnSuccessCallback($successCustom = '', $checkoutType)
+    {
+        $saveOrderUrl = $this->getUrl('boltpay/order/save');
+
+        if ($checkoutType === self::CHECKOUT_TYPE_ADMIN) {
+            $onSuccessCallback = "function(transaction, callback) {
```

Comment:
you don't need `$onSuccessCallback`. you can just return.

Reply:
@daisy1754  Are you sure? Because it was there previously.

Reply:
@YevhenShBolt what I mean by my comment is you can write:

```
if ($checkoutType === self::CHECKOUT_TYPE_ADMIN) {
    return "function(transaction, callback) {
        $successCustom
        var input = document.createElement('input');
        input.setAttribute('type', 'hidden');
        input.setAttribute('name', 'bolt_reference');
        input.setAttribute('value', transaction.reference);
        document.getElementById('edit_form').appendChild(input);
        // order and order.submit should exist for admin
        if ((typeof order !== 'undefined' ) && (typeof order.submit === 'function')) {
            order_completed = true;
            callback();
        }
    }";
} 
```

No need for variable `$onSuccessCallback`